internal/controller/csiaddons/networkfence_controller.go

```diff
@@ -59,6 +59,16 @@ func validateNetworkFenceSpec(nwFence *csiaddonsv1alpha1.NetworkFence) error {
 	if nwFence == nil {
 		return errors.New("NetworkFence resource is empty")
 	}
+
+	if nwFence.Spec.NetworkFenceClassName != "" {
+		if nwFence.Spec.Cidrs == nil {
+			return errors.New("required parameter cidrs is not specified")
+		}
+
+		// Driver name and secrets will be read (and validated) in NetworkFenceClass
+		return nil
+	}
+
 	if nwFence.Spec.Driver == "" {
```

Comment:
You need to skip validation for Driver if the class is specified.

Reply:
It will be skipped using the check just above, we return early if CIDRs are present and NFClassName is specified.